ibis/backends/base/sql/compiler/query_builder.py

```diff
@@ -41,12 +41,19 @@ def __init__(self, parent, node, indent=2):
         self.indent = indent
 
         self.join_tables = []
```

Comment:
Responding to https://github.com/ibis-project/ibis/pull/7894#discussion_r1441543842:

Good point. I assumed that the addition of `record_join_tables()` and `alias` arg for `_format_table()` would help the logic for all backends but I surely cannot make the call on that :)

@cpcloud What do you think about this change in `query_builder.py`?